controllers/helpers.go

```diff
@@ -67,8 +70,11 @@ func deleteVM(ctx context.Context, client *nutanixClientV3.Client, vmName, vmUUI
 }
 
 // findVMByUUID retrieves the VM with the given vm UUID. Returns nil if not found
-func findVMByUUID(ctx context.Context, client *nutanixClientV3.Client, uuid string) (*nutanixClientV3.VMIntentResponse, error) {
+func findVMByUUID(ctx context.Context, client *nutanixClientV3.Client, uuid, expectedName string) (*nutanixClientV3.VMIntentResponse, error) {
 	klog.Infof("Checking if VM with UUID %s exists.", uuid)
+	if expectedName == "" {
+		return nil, fmt.Errorf("expectedName required to be set when searching for VM by UUID")
+	}
```

Comment:
imo, expected name comparison shouldn't be part of this function.

Reply:
Changed